src/backend/main.ml

```diff
@@ -4,7 +4,12 @@
 (* Helper functions *)
 (* ************************************************************************* *)
 
-let loader _root path _request =
+let index_loader _root _path _request =
+  match Static.read "index.html" with
+  | Some index -> Dream.html index
+  | None -> Dream.respond ~status:`Not_Found "Index file not found"
```

Comment:
Nope, this will again, only ever serve `index.html`. What I meant was use the following code:
```ocaml
match Static.read path with
  | None ->
    (* if the path is not found in the frontend, automatically redirect to `index.html` *)
    begin match Static.read "index.html" with
        | None -> assert false (* let's assume the frontend will always have an `index.html` *)
        | Some asset -> Dream.respond asset
      end
  | Some asset -> Dream.respond asset
```